### PR TITLE
Fix AutoComplete behaviour

### PR DESCRIPTION
**Pull request by niconoir**

### CODE REVIEW

**Review by laurgarn — APPROVED** [review]

File: src/components/dialogs/equipment-deletion-dialog.js [hunk 1]
```diff
@@ -71,19 +111,29 @@ const EquipmentDeletionDialog = ({
...
-    const [searchMatchingEquipments, equipmentsFound] =
-        useSearchMatchingEquipments(
-            studyUuid,
-            currentNodeUuid,
-            true,
-            equipmentType,
-            makeItems
-        );
+    useEffect(() => {
+        setEquipmentsFound([]);
```
Comment (etiennehomer) [inline src/components/dialogs/equipment-deletion-dialog.js:122]: We can avoid this set ?

File: src/components/dialogs/equipment-deletion-dialog.js [hunk 2]
```diff
@@ -189,15 +238,20 @@ const EquipmentDeletionDialog = ({
...
-                            {equipmentTypes.map((item) => {
-                                return (
-                                    <MenuItem key={item} value={item}>
-                                        {intl.formatMessage({
-                                            id: item,
-                                        })}
-                                    </MenuItem>
-                                );
-                            })}
+                            {Object.entries(equipmentTypes).map(
```
Comment (etiennehomer) [inline src/components/dialogs/equipment-deletion-dialog.js:241]: Object.values 
id is not used

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/components/dialogs/equipment-deletion-dialog.js [hunk 3]
```diff
@@ -71,19 +111,29 @@ const EquipmentDeletionDialog = ({
...
-        );
+    useEffect(() => {
+        setEquipmentsFound([]);
+        if (Array.isArray(equipmentType.fetcher)) {
+            Promise.all(
+                equipmentType.fetcher.map((fetchPromise) =>
+                    fetchPromise(studyUuid, currentNodeUuid)
+                )
+            ).then((vals) => {
+                setEquipmentsFound(vals?.flat().sort(compareById));
```
Comment (SlimaneAmar) [inline src/components/dialogs/equipment-deletion-dialog.js:129]: vals may be null ?

File: src/components/dialogs/equipment-deletion-dialog.js (hunk 1 above)
Comment (niconoir) [reply src/components/dialogs/equipment-deletion-dialog.js:122]: If we remove this set, a glitch can appear when displaying equipments list in the autocomplete (when moving from one equipment type to another)

File: src/components/dialogs/equipment-deletion-dialog.js (hunk 2 above)
Comment (niconoir) [reply src/components/dialogs/equipment-deletion-dialog.js]: Right, corrected.

File: src/components/dialogs/equipment-deletion-dialog.js [hunk 4]
```diff
@@ -4,53 +4,93 @@
...
-        .map((e) => {
-            let label = usesNames ? e.name : e.id;
-            return {
-                label: label,
-                id: e.id,
-                key: e.id,
-            };
-        })
-        .sort((a, b) => a.label.localeCompare(b.label));
+const equipmentTypes = {
```
[the remark this answers is not in the document]
Comment (niconoir) [reply src/components/dialogs/equipment-deletion-dialog.js:38]: Currently only used in (and specific to) equipment-deletion-dialog

File: src/components/dialogs/equipment-deletion-dialog.js (hunk 3 above)
Comment (niconoir) [reply src/components/dialogs/equipment-deletion-dialog.js]: You're right, `vals` should not be null (removed the `?`). Also added a catch to handle fetching errors.

**Review by SlimaneAmar — APPROVED** [review]

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**